### DISCUSSION TITLE
Cookie collection via window browsing

### DISCUSSION THREAD

**Discussion by mrudelle** [Kong/insomnia#3634] [Ideas]
- Insomnia Version: 5.12.4.1752
- Operating System: OSX 10.11.6

## Details

The idea is to allow to easily collect cookies (eg. for authentication) by allowing the user to manually browse a specific url. 

The need arises when using the Authorization Code grant type of OAuth2.0 to authenticate users with the server. Insomnia doesn't need to be configured specially for this authentication scheme like it is enforceds ATM. And ideally the user would just specify a route to be visited via a browser window, the authentication flow would set all the required cookies such that subsequent requests are authenticated.

## More notes

The option could be available in the "cookies" pane, and would allow the user to visit a specific url. Then insomnia would collect cookies set during the interaction (perhaps limiting collection to some domains or specific cookies in order to avoid polluting the workspace).

**Comment by welcome** [comment]
👋 Thanks for opening your first issue! If you're reporting a 🐞 bug, please make sure
you include steps to reproduce it. If you're requesting a feature 🎁, please provide real
use cases that would benefit. 👪

To help make this a smooth process, please be sure you have first read the
[contributing guidelines](https://github.com/getinsomnia/insomnia/blob/master/CONTRIBUTING.md).

**Comment by gschier** [comment]
That sounds like it could be really useful. Would you be interested in helping implement this feature?

**Comment by mrudelle** [comment]
@gschier I'll try to draft an implementation. Right now I'm facing a libcurl version error when running the app, I'll look into it later.
